source/dub/dependency.d

```diff
@@ -829,4 +966,21 @@ unittest {
 	assertThrown(a == b, "Failed: UNKNOWN == UNKNOWN");
 
 	assert(Version("1.0.0+a") == Version("1.0.0+b"));
+
+	assert(Version("73535568b79a0b124bc1653002637a830ce0fcb8").isSCM);
+}
+
+/// Determines whether the given string is a Git hash.
+bool isGitHash(string hash) @nogc nothrow pure @safe
+{
+	import std.ascii : isAlphaNum;
+	import std.utf : byCodeUnit;
+
+	return hash.length >= 7 && hash.length <= 40 && hash.byCodeUnit.all!isAlphaNum;
```

Comment:
this should be `isHexDigit` (as commit hashes are hex encoded)

If we want to support tag names I would instead suggest adding some kind of additional prefixed syntax in a later PR

Reply:
True that.